packages/badge/src/Badge.ts

```diff
@@ -35,21 +37,68 @@ export const BADGE_VARIANTS = [
     'yellow',
 ] as const;
 export type BadgeVariant = typeof BADGE_VARIANTS[number];
+export const FIXED_VALUES_DEPRECATED = ['top', 'bottom', 'left', 'right'];
+export const FIXED_VALUES = [
+    'inline-start',
+    'inline-end',
+    'block-start',
+    'block-end',
+] as const;
+export type FixedValues =
+    | typeof FIXED_VALUES[number]
+    | typeof FIXED_VALUES_DEPRECATED[number];
 
 /**
  * @element sp-badge
  *
  * @slot - Text label of the badge
  * @slot icon - Optional icon that appears to the left of the label
  */
-export class Badge extends SizedMixin(ObserveSlotText(SpectrumElement, '')) {
+export class Badge extends SizedMixin(
+    ObserveSlotText(ObserveSlotPresence(SpectrumElement, '[slot="icon"]'), '')
+) {
     public static override get styles(): CSSResultArray {
         return [styles];
     }
 
+    @property({ reflect: true })
+    public get fixed(): FixedValues | undefined {
+        return this._fixed;
+    }
+
+    public set fixed(fixed: FixedValues | undefined) {
+        if (fixed === this.fixed) return;
+        const oldValue = this.fixed;
+        if (window.__swc.DEBUG) {
+            if (fixed && FIXED_VALUES_DEPRECATED.includes(fixed)) {
+                window.__swc.warn(
+                    this,
+                    `The following values for "fixed" in <${this.localName}> have been deprecated. They will be removed in a future release.`,
+                    'https://opensource.adobe.com/spectrum-web-components/components/badge/#fixed',
+                    {
+                        issues: [...FIXED_VALUES_DEPRECATED],
+                    }
+                );
+            }
+        }
```

Comment:
👍 on warning, I need to start looking for places to provide dev mode hints myself.

Once we start sprinkling in more than one or two, it might be nice to follow a pattern like:

```ts
this.warnIfDeprecatedFixed(fixed);

// ...

protected warnIfDeprecatedFixed(fixed: FixedValues | undefined) {
    if (!window.__swc.DEBUG) return;
    if (!(fixed && FIXED_VALUES_DEPRECATED.includes(fixed))) return;

    window.__swc.warn(this, '...', { issues: [...FIXED_VALUES_DEPRECATED] });
}
```

Reply:
This feels like this nicer way to do this, _BUT_ if you write it like this you'd have to write custom tooling to get this to be pulled out of the "for production" files. Wrapping all warn code in `if (window.__swc.DEBUG) {` means that when we convert `window.__swc.DEBUG` to `false` in esbuild the subsequent terser passes will eliminate the dead code for us.

Reply:
Aha! Good to know. I may try something like `if (window.__swc.DEBUG) this.warnIfDeprecatedFixed(fixed);` then.

Reply:
Interesting, whether it pulls out the unused `private` would be a good test. Look forward to hearing what you find!